multitenancy/src/main/java/org/axonframework/extensions/multitenancy/components/deadletterqueue/MultiTenantDeadLetterQueue.java

```diff
@@ -0,0 +1,406 @@
+     *
+     * @return a Builder to be able to create a {@link MultiTenantDeadLetterQueue}.
+     */
+    public static MultiTenantDeadLetterQueue.Builder builder() {
+        return new MultiTenantDeadLetterQueue.Builder();
+    }
+
+    /**
+     * Registers a {@link SequencedDeadLetterQueue} that will be used by tenants.
+     *
+     * @param deadLetterQueue the {@link SequencedDeadLetterQueue} that will be used by tenants.
+     */
+    public void registerDeadLetterQueueSupplier(Supplier<SequencedDeadLetterQueue<M>> deadLetterQueue) {
+        deadLetterQueueSupplier = deadLetterQueue;
+    }
+
+    /**
+     * Gets the {@link SequencedDeadLetterQueue} for the given {@link TenantDescriptor}.
+     * If the tenant is not registered, it will return null. If the tenant is registered, but the
+     * {@link SequencedDeadLetterQueue} is not yet created, it will create it and return it.
+     */
+    public SequencedDeadLetterQueue<M> getTenantSegment(TenantDescriptor tenantDescriptor) {
+        return tenantSegments.computeIfAbsent(tenantDescriptor, t -> {
+            if (tenants.contains(tenantDescriptor)) {
+                return deadLetterQueueSupplier.get();
+            }
+            return null;
+        });
+    }
+
+    private SequencedDeadLetterQueue<M> resolveTenant(DeadLetter<? extends M> deadLetter) {
+        TenantDescriptor tenantDescriptor = targetTenantResolver.resolveTenant(deadLetter.message(),
+                                                                               tenantSegments.keySet());
+        SequencedDeadLetterQueue<M> tenantDeadLetterQueue = getTenantSegment(tenantDescriptor);
+        if (tenantDeadLetterQueue == null) {
+            throw new NoSuchTenantException(tenantDescriptor.tenantId());
+        }
+        return tenantDeadLetterQueue;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void enqueue(Object sequenceIdentifier, DeadLetter<? extends M> letter)
+            throws DeadLetterQueueOverflowException {
+        resolveTenant(letter).enqueue(sequenceIdentifier, letter);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean enqueueIfPresent(Object sequenceIdentifier, Supplier<DeadLetter<? extends M>> letterBuilder)
+            throws DeadLetterQueueOverflowException {
+        return resolveTenant(letterBuilder.get()).enqueueIfPresent(sequenceIdentifier, letterBuilder);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void evict(DeadLetter<? extends M> letter) {
+        resolveTenant(letter).evict(letter);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void requeue(DeadLetter<? extends M> letter, UnaryOperator<DeadLetter<? extends M>> letterUpdater)
+            throws NoSuchDeadLetterException {
+        resolveTenant(letter).requeue(letter, letterUpdater);
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
I think we should cover the if/else-statement for all methods down below, as it is differing behavior depending on how these operations are invoked.